chore: fixes the upgrade tests to pass ci

cmwylie19 opened the pull request:
## Description

Fixes the upgrade test in order to pass CI. The upgrade builds a Pepr module with latest, deploys it, then rebuilds with the pepr release candidate. In the v0.51.0 there will be a migration path that must occur for this behavior to happen due to a breaking change in ESLint. The upgrade path is documented [here](https://github.com/defenseunicorns/pepr-excellent-examples/pull/298). For now, we are using `nightly` in place of latest in order to test the upgrade path in the same manner as before. This change will need to be updated after the v0.51.0 release 

## Related Issue

Fixes #2152
<!-- or -->
Relates to #2153

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

cmwylie19 committed:
chore: journey to use nightly

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
chore: change upgrade test to pass ci using nightly

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
chore: clean up code

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

samayer12 reviewed (approved):
Good workaround until the next release. Added #2155 to remind us to back this change out after we're more stable.

cmwylie19 commented:
> Good workaround until the next release. Added #2155 to remind us to back this change out after we're more stable.

Ah thank you! I had added this #2153 but I will close in favor of this issue.
